ui/pages/confirmations/send/send-content/add-recipient/domain-input-resolution-cell.js

```diff
@@ -0,0 +1,200 @@
+import React, {
+  ///: BEGIN:ONLY_INCLUDE_IF(build-flask)
+  useContext,
+  ///: END:ONLY_INCLUDE_IF
+  useRef,
+  useEffect,
+  useState,
+} from 'react';
+import PropTypes from 'prop-types';
+///: BEGIN:ONLY_INCLUDE_IF(build-flask)
+import { I18nContext } from '../../../../../contexts/i18n';
+///: END:ONLY_INCLUDE_IF
+import Identicon from '../../../../../components/ui/identicon';
+import Confusable from '../../../../../components/ui/confusable';
+import {
+  ///: BEGIN:ONLY_INCLUDE_IF(build-flask)
+  AvatarIcon,
+  BadgeWrapper,
+  IconName,
+  IconSize,
+  ///: END:ONLY_INCLUDE_IF
+  Text,
+} from '../../../../../components/component-library';
+import {
+  ///: BEGIN:ONLY_INCLUDE_IF(build-flask)
+  BackgroundColor,
+  IconColor,
+  ///: END:ONLY_INCLUDE_IF
+  TextColor,
+  TextVariant,
+} from '../../../../../helpers/constants/design-system';
+import { ellipsify } from '../../send.utils';
+import Tooltip from '../../../../../components/ui/tooltip';
+
+export default function DomainInputResolutionCell({
+  domainType,
+  address,
+  domainName,
+  ///: BEGIN:ONLY_INCLUDE_IF(build-flask)
+  resolvingSnap,
+  ///: END:ONLY_INCLUDE_IF
+  onClick,
+  protocol,
+}) {
+  ///: BEGIN:ONLY_INCLUDE_IF(build-flask)
+  const t = useContext(I18nContext);
+  ///: END:ONLY_INCLUDE_IF
+  const titleRef = useRef(null);
+  const breakpointRef = useRef(null);
+  const [isTitleOverflowing, setIsTitleOverflowing] = useState(false);
+
+  useEffect(() => {
+    let isOverflowing =
+      titleRef.current.offsetWidth < titleRef.current.scrollWidth;
+    const breakpointLength = titleRef.current.textContent.length;
+
+    if (isOverflowing && !breakpointRef.current) {
+      breakpointRef.current = breakpointLength;
+    }
+
+    if (!isOverflowing) {
```

Comment:
What is going on here

Reply:
This logic is to ensure that the cell properly shows the ellipsis in the domain resolution cell